### PR TITLE
feat: Add support for deleting TPW items

### LINKED ISSUE #318: Add support for deleting Third Party Items
Third party items can be deleted only if they were not published before.
 To complete this task:
- Add the checks to prevent the items from being deleted if they live in the TPR.

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds support for deleting TPW items.
Closes #318

### CODE REVIEW

File: src/Collection/utils.ts [hunk 1]
```diff
@@ -24,6 +24,10 @@ export function getThirdPartyCollectionURN(
   return `${third_party_id}:${urn_suffix}`
 }
 
+export function isTPCollection(collection: CollectionAttributes): boolean {
```
Comment (nicosantangelo) [inline src/Collection/utils.ts:27]: We can use this method on `upsertTPWCollection` too!

```ts
if (collection) {
if (
  collection.third_party_id === null ||
  collection.urn_suffix === null
) {
  throw new WrongCollectionException(
    "The collection can't be converted into a third party collection.",
    { id }
  )
}
```

File: src/Item/Item.exceptions.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+}
+
+export class CollectionForItemLockedException extends Error {
+  constructor(public id: string, action: ItemAction) {
+    super(`The collection for the item is locked. The item can't be ${action}.`)
+  }
+}
+
+export class InconsistentItemException extends Error {
```
Comment (nicosantangelo) [inline src/Item/Item.exceptions.ts:41]: I know we already did this on `Collection.exceptions.ts` but what do you think of renaming all occurrences of `Exception` to `Error` ?

It's not wrong by any means, it's just that the concept in JS is called `Error`

File: src/Item/utils.ts [hunk 3]
```diff
@@ -25,3 +30,27 @@ export function toDBItem(item: FullItem): ItemAttributes {
...
+  const matches = tpwItemURNRegex.exec(urn)
+  if (matches === null || matches.length !== 2) {
+    throw new Error('The given collection URN is not TWP compliant')
+  }
+
+  return { collectionId: matches[0], itemId: matches[1] }
+}
+
+export function isTPItem(item: ItemAttributes): boolean {
+  return item.urn_suffix !== null
```
Comment (nicosantangelo) [inline src/Item/utils.ts:55]: Can/Should we use this on `isOwnedOrManagedBy` on `ItemService`?
```ts
} else if (dbItem.urn_suffix && dbItem.collection_id) {
  return this.collectionService.isOwnedOrManagedBy(
    dbItem?.collection_id,
    ethAddress
  )
} else {
```

**Review by nicosantangelo — COMMENTED** [review]
Gr8

File: src/Collection/utils.ts (hunk 1 above)
Comment (LautaroPetaccio) [reply src/Collection/utils.ts:27]: Used the `isTPCollection` function here.

File: src/Item/Item.exceptions.ts (hunk 2 above)
Comment (LautaroPetaccio) [reply src/Item/Item.exceptions.ts]: Changed Exception to Error only in the items.

File: src/Item/utils.ts (hunk 3 above)
Comment (LautaroPetaccio) [reply src/Item/utils.ts]: We should, yes! Added it here.

**Review by nicosantangelo — APPROVED** [review]

**Pull request merged**